huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_cluster_test.go

```diff
@@ -0,0 +1,672 @@
+}
+`, rName, rName)
+}
+
+// The task node has not contain data disks.
+func testAccMrsMapReduceClusterConfig_basic(rName, pwd string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_mapreduce_cluster" "test" {
+  availability_zone  = data.huaweicloud_availability_zones.test.names[0]
+  name               = "%s"
+  type               = "STREAMING"
+  version            = "MRS 1.9.2"
+  manager_admin_pass = "%s"
+  node_admin_pass    = "%s"
+  subnet_id          = huaweicloud_vpc_subnet.test.id
+  vpc_id             = huaweicloud_vpc.test.id
+  component_list     = ["Storm"]
+
+  master_nodes {
+    flavor            = "c6.2xlarge.4.linux.bigdata"
+    node_number       = 2
+    root_volume_type  = "SAS"
+    root_volume_size  = 300
+    data_volume_type  = "SAS"
+    data_volume_size  = 480
+    data_volume_count = 1
+  }
+  streaming_core_nodes {
+    flavor            = "c6.2xlarge.4.linux.bigdata"
+    node_number       = 2
+    root_volume_type  = "SAS"
+    root_volume_size  = 300
+    data_volume_type  = "SAS"
+    data_volume_size  = 480
+    data_volume_count = 1
+  }
+  streaming_task_nodes {
+    flavor            = "c6.2xlarge.4.linux.bigdata"
+    node_number       = 1
+    root_volume_type  = "SAS"
+    root_volume_size  = 300
+  }
+
+  tags = {
+    foo = "bar"
+    key = "value"
+  }
+}`, testAccMrsMapReduceClusterConfig_base(rName), rName, pwd, pwd)
+}
+
+func testAccMrsMapReduceClusterConfig_update(rName, pwd string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_mapreduce_cluster" "test" {
+  availability_zone  = data.huaweicloud_availability_zones.test.names[0]
+  name               = "%s"
+  type               = "STREAMING"
+  version            = "MRS 1.9.2"
+  manager_admin_pass = "%s"
+  node_admin_pass    = "%s"
+  subnet_id          = huaweicloud_vpc_subnet.test.id
+  vpc_id             = huaweicloud_vpc.test.id
+  component_list     = ["Storm"]
+
+  master_nodes {
+    flavor            = "c6.2xlarge.4.linux.bigdata"
+    node_number       = 2
+    root_volume_type  = "SAS"
+    root_volume_size  = 300
+    data_volume_type  = "SAS"
+    data_volume_size  = 480
+    data_volume_count = 1
+  }
+  streaming_core_nodes {
+    flavor            = "c6.2xlarge.4.linux.bigdata"
+    node_number       = 2
+    root_volume_type  = "SAS"
+    root_volume_size  = 300
+    data_volume_type  = "SAS"
+    data_volume_size  = 480
+    data_volume_count = 1
+  }
+  streaming_task_nodes {
+    flavor            = "c6.2xlarge.4.linux.bigdata"
+    node_number       = 1
+    root_volume_type  = "SAS"
+    root_volume_size  = 300
+  }
+
+  tags = {
+    foo1 = "bar"
+    key  = "update_value"
+  }
+}`, testAccMrsMapReduceClusterConfig_base(rName), rName, pwd, pwd)
+}
+
+func testAccMrsMapReduceClusterConfig_keypair(rName, pwd string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_compute_keypair" "test" {
+  name = "%s"
+
+  lifecycle {
+    ignore_changes = [
+      public_key,
+    ]
+  }
```

Comment:
why we add `lifecycle` block here?

Reply:
after applying terraform configuration, the plan will be not empty (public_key is a ForceNew parameter).